src/main.cc

```diff
@@ -1,9 +1,36 @@
-#include "pdtt.hh"
+#include <iostream>
 
-auto main()
-    -> int
-{
-    PDTT pdtt;
-    pdtt.run();
+#include "Arguments.h"
+#include "Interactor.h"
+#include "StandardScreen.h"
+#include "Trainer.h"
+
+int main(int argc, char** argv) {
+  Arguments args(argc, argv);
+
+  auto is_help = args.is_help();
+  if (is_help) {
+    std::cout << args.GetHelp();
     return 0;
+  }
+
+  auto filename = args.file_path();
+  if (!filename) {
+    std::cout << "You have to provide filename.\n -h to help.\n";
+    return 1;
+  }
+
+  auto n_words = args.n_words();
+  if (!n_words) {
+    std::cout
+        << "You have to provide number of words to train on.\n -h to help.\n";
+    return 1;
+  }
+
+  StandardScreen::StartNcursesMode();
```

Comment:
Maybe code that starts here should be separated in other function or class like `App`

Reply:
moved to external function